### PR TITLE
Add Python 3.6 to CI

### PR DESCRIPTION
**Pull request by derekbekoe**
Closes https://github.com/Azure/azure-cli/issues/1626

### CODE REVIEW

File: scripts/build_in_travis.sh [hunk 1]
```diff
@@ -12,7 +12,10 @@ scripts_root=$(cd $(dirname $0); pwd)
 export PYTHONPATH=$PATHONPATH:./src
 python -m azure.cli -h
 
-check_style --ci
+# PyLint does not yet support Python 3.6 https://github.com/PyCQA/pylint/issues/1241
+if [ "$TRAVIS_PYTHON_VERSION" != "3.6" ]; then
+    check_style --ci;
+fi
```
Comment (troydai) [inline scripts/build_in_travis.sh]: yeah, super annoy.

**Comment by derekbekoe** [comment]
@troydai can you take a look at the updated PR?

**Comment by troydai** [comment]
@derekbekoe  On it.

File: src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py [hunk 2]
```diff
@@ -46,6 +46,10 @@
 
 # MOCK METHODS
 
+# Workaround until https://github.com/kevin1024/vcrpy/issues/293 is fixed.
+vcr_connection_request = vcr.stubs.VCRConnection.request
+vcr.stubs.VCRConnection.request = lambda *args, **kwargs: vcr_connection_request(*args)
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py]: I'm pretty sure flake8 will bad mouth this line of code. But we disable the style check here anyway.

**Pull request merged**